src/pip/_internal/commands/debug.py

```diff
@@ -43,6 +47,102 @@ def show_sys_implementation():
         show_value('name', implementation_name)
 
 
+def find_vendor_txt_path():
+    # type: () -> str
+    vendor_txt_path = os.path.join(
+        os.path.dirname(pip_location),
+        '_vendor',
+        'vendor.txt'
+    )
+    assert os.path.exists(vendor_txt_path), \
+        'Could not find vendor.txt file!'
+
+    return vendor_txt_path
+
+
+def create_vendor_txt_map():
+    # type: () -> Dict[str, str]
+    vendor_txt_path = find_vendor_txt_path()
+
+    with open(vendor_txt_path) as f:
+        # Purge non version specifying lines.
+        # Also, remove any space prefix or suffixes (including comments).
+        lines = [line.strip().split(' ', 1)[0]
+                 for line in f.readlines() if '==' in line]
+
+    # Transform into "module" -> version dict.
+    return dict(line.split('==', 1) for line in lines)  # type: ignore
+
+
+def get_module_from_module_name(module_name):
+    # type: (str) -> ModuleType
+
+    # Module name can be uppercase in vendor.txt for some reason...
+    module_name = module_name.lower()
+    # PATCH: setuptools is actually only pkg_resources.
+    if module_name == 'setuptools':
+        module_name = 'pkg_resources'
+
+    __import__(
+        'pip._vendor.{}'.format(module_name),
+        globals(),
+        locals(),
+        level=0
+    )
+    return getattr(pip._vendor, module_name)
+
+
+def get_vendor_version_from_module(module_name):
+    # type: (str) -> str
+
+    module = get_module_from_module_name(module_name)
+    version = getattr(module, '__version__', None)
+    if isinstance(version, ModuleType):
```

Comment:
Why is this needed ?

Reply:
Added now as a comment
    # Some vendored modules (like requests) have a __version__.py file.
    # If this is the case, we should grab __version__
    # from the __version__.py file